File: alumnium/tools/press_key_tool.py
from enum import Enum

from pydantic import BaseModel, Field

from alumnium.drivers import PlaywrightDriver, SeleniumDriver


class Key(str, Enum):
    BACKSPACE = "Backspace"
    ENTER = "Enter"
    ESCAPE = "Escape"
    TAB = "Tab"


class PressKeyTool(BaseModel):
    """Presses a key on the keyboard."""

    key: Key = Field(description="Key to press.")

    def invoke(self, driver: PlaywrightDriver | SeleniumDriver):
        driver.press_key(self.key)

File: ts/src/tools/PressKeyTool.ts
import { BaseDriver } from "../drivers/BaseDriver.js";
import { Key } from "../drivers/keys.js";
import { BaseTool } from "./BaseTool.js";

export class PressKeyTool extends BaseTool {
  static description = "Press a keyboard key.";

  key: Key;

  constructor(args: { key: Key }) {
    super();
    this.key = args.key;
  }

  async invoke(driver: BaseDriver): Promise<void> {
    await driver.pressKey(this.key);
  }
}
